app/Jobs/FetchCollectionActivity.php

```diff
@@ -0,0 +1,125 @@
+
+    /**
+     * Create a new job instance.
+     */
+    public function __construct(
+        public Collection $collection,
+        public bool $forced = false,
+    ) {
+        $this->onQueue(Queues::SCHEDULED_NFTS);
+    }
+
+    /**
+     * Execute the job.
+     */
+    public function handle(MnemonicWeb3DataProvider $provider): void
+    {
+        if ($this->collection->isSpam()) {
+            return;
+        }
+
+        if ($this->collection->is_fetching_activity && ! $this->forced) {
+            return;
+        }
+
+        $this->collection->update([
+            'is_fetching_activity' => true,
+        ]);
+
+        $limit = 500;
+
+        $activities = $provider->getCollectionActivity(
+            chain: $this->collection->network->chain(),
+            contractAddress: $this->collection->address,
+            limit: $limit,
+            from: $this->latestActivityTimestamp(),
+        );
+
+        if ($activities->isEmpty()) {
+            $this->collection->update([
+                'is_fetching_activity' => false,
+                'activity_updated_at' => now(),
+            ]);
+
+            return;
+        }
+
+        $formattedActivities = $activities
+            // Sometimes the request is returning transfers that are not labeled as any of the values we expect.
+            // There were times when Mnemonic returned `LABEL_BURN` transfers, so we're filtering them here.
+            ->reject(fn ($activity) => $activity->type === null)
+            ->unique->key()
+            ->map(fn (CollectionActivity $activity) => [
+                'collection_id' => $this->collection->id,
+                'token_number' => $activity->tokenId,
+                'type' => $activity->type->value,
+                'sender' => $activity->sender,
+                'recipient' => $activity->recipient,
+                'tx_hash' => $activity->txHash,
+                'timestamp' => $activity->timestamp,
+                'total_native' => $activity->totalNative,
+                'total_usd' => $activity->totalUsd,
+                'extra_attributes' => json_encode($activity->extraAttributes),
+            ])->toArray();
+
+        DB::transaction(function () use ($formattedActivities, $activities, $limit) {
+            NftActivity::upsert($formattedActivities, uniqueBy: ['tx_hash', 'collection_id', 'token_number', 'type']);
+
+            // If we get the limit it may be that there are more activities to fetch...
+            if ($limit === count($activities)) {
+                self::dispatch($this->collection, forced: true)->afterCommit();
+            } else {
+                $this->collection->update([
+                    'is_fetching_activity' => false,
+                    'activity_updated_at' => now(),
+                ]);
+            }
+        }, attempts: 5);
+    }
+
+    private function latestActivityTimestamp(): ?Carbon
+    {
+        return $this->collection
+                    ->activities()
```

Comment:
```suggestion
                    ->activities()
                    ->select('timestamp')
```

in theory that should make the query faster

Reply:
interestingly this produces the same SQL query :noice:

```
> Collection::find(3)->activities()->latest('timestamp')->select('timestamp')->value('timestamp')
= select "timestamp" from "nft_activity" where "nft_activity"."collection_id" = ? and "nft_activity"."collection_id" is not null order by "timestamp" desc limit 1  

> Collection::find(3)->activities()->latest('timestamp')->value('timestamp')
= select "timestamp" from "nft_activity" where "nft_activity"."collection_id" = ? and "nft_activity"."collection_id" is not null order by "timestamp" desc limit 1  

```